worker/games.py

```diff
@@ -439,7 +446,7 @@ def setup_engine(
 
         shutil.move("stockfish" + EXE_SUFFIX, destination)
     except:
-        raise Exception("Failed to setup engine for {}".format(sha))
+        raise WorkerException("Failed to setup engine for {}".format(sha))
```

Comment:
This is already bad in master because masks all the inner exceptions, I was working on this problem with a test `raise`:
```python
raise WorkerException("Failed to validate the network: {}".format(net)) # TEST
```
```
Build uses default net:  nn-13406b1dcbe0.nnue

Exception running games:
Failed to setup engine for c27b5be9e015bd9f9f76d471a9904e5ad83bae15
Informing the server

```
Solutions:
1. forward the inner exception and print a message:
```python
    except Exception as e:
        print ("Failed to setup engine for {}".format(sha))
        raise
```

```
Build uses default net:  nn-13406b1dcbe0.nnue
Failed to setup engine for c27b5be9e015bd9f9f76d471a9904e5ad83bae15

Exception running games:
Failed to validate the network: nn-13406b1dcbe0.nnue
Informing the server
Task exited
```

2. print the inner exception and raise the outer exception:
```python
    except Exception as e:
        print ("Exception during the engine setup:\n", e, sep="", file=sys.stderr)
        raise WorkerException("Failed to setup engine for {}".format(sha))
```
```
Build uses default net:  nn-13406b1dcbe0.nnue
Exception during engine setup:
Failed to validate the network: nn-13406b1dcbe0.nnue

Exception running games:
Failed to setup engine for c27b5be9e015bd9f9f76d471a9904e5ad83bae15
Informing the server
Task exited
```

3. raise the tuple with outer and inner exceptions:
```python
    except Exception as e:
        raise WorkerException("Failed to setup engine for {}".format(sha), e)
```
```
Build uses default net:  nn-13406b1dcbe0.nnue

Exception running games:
('Failed to setup engine for c27b5be9e015bd9f9f76d471a9904e5ad83bae15', WorkerException('Failed to validate the network: nn-13406b1dcbe0.nnue'))
Informing the server
Task exited
```
This is the "Actions" entry for the last point
<html>
<body>
<!--StartFragment-->

21-10-16 14:40:13 | user00 | sbvals1 | failed task: task_id: 215, worker: user00-3cores-e54411c4, reason: '('Failed to setup engine for c27b5be9e015bd9f9f76d471a9904e5ad83bae15', WorkerException('Terminated by signal'))'
-- | -- | -- | --
21-10-16 14:39:57 | user00 | sbvals1 | failed task: task_id: 214, worker: user00-3cores-e54411c4, reason: '('Failed to setup engine for c27b5be9e015bd9f9f76d471a9904e5ad83bae15', WorkerException('Failed to validate the network: nn-13406b1dcbe0.nnue'))'

<!--EndFragment-->
</body>
</html>

Reply:
My preference would be for the first solution. I am a bit worried about cluttering the action page too much....

Reply:
Also this works and perhaps is the best solution (the string can be formatted in any way):
```python
    except Exception as e:
        raise WorkerException("Failed to setup engine for {}:\n{}".format(sha, e))
```
```
Build uses default net:  nn-13406b1dcbe0.nnue

Exception running games:
Failed to setup engine for c27b5be9e015bd9f9f76d471a9904e5ad83bae15:
Failed to validate the network: nn-13406b1dcbe0.nnue
Informing the server
```

Reply:
> My preference would be for the first solution. I am a bit worried about cluttering the action page too much....

OK, lets go!

Reply:
But then I am again worried about leaking information if `e` is e.g. a syntax error.

Reply:
I think the inner exceptions are the most important ones since they indicate where the error is. So I agree that those should be preserved. The outer exceptions have a more cosmetic role.

Reply:
With 1. an error from other code within the `try:` is reported at the `setup_engine()` call `failed task: task_id: 262, worker: user00-3cores-92f1b2f4, reason: 'Exception at games.py: 922'`